Assets/Features/Building/Scripts/Grid/Grid.cs

```diff
@@ -421,14 +355,14 @@ public bool Remove(Vector3Int gridVector)
                     }
 
                     _cells[item.x, item.y, item.z].Clear();
+                    
+                    GameManager.Instance.EventManager.TriggerEvent(EventId.OnChangeTile, new TileUpdateData()
```

Comment:
Seems like this event is also listened to within the Grid functionality. This is not very common and not a common practice